element_calcium_imaging/imaging.py

```diff
@@ -119,6 +120,29 @@ class ProcessingTask(dj.Manual):
     task_mode='load': enum('load', 'trigger')   # 'load': load computed analysis results, 'trigger': trigger computation
     """
 
+    @classmethod
+    def auto_generate_entries(cls, scan_key, task_mode):
+        """
+        Method to auto-generate ProcessingTask entries for a particular Scan.
+        """
+
+        image_locators = [get_scan_image_files, get_scan_box_files, get_nd2_files]
+        for image_locator in image_locators:
+            try:
+                relative_scan_dir = pathlib.Path(image_locator(scan_key)[0]).relative_to(get_imaging_root_data_dir()).parent
+            except:
+                pass
+            if relative_scan_dir:
+                break
```

Comment:
You don't need to do this loop, from `scan_key` you would know the acquisition software, and you can call the corresponding `image_locator` there.

Reply:
Right! I'll use a dictionary